GDAL with ECW for macos link failure

### What is the bug?

Building with ECW 5.5 driver for macos fails with

```
Undefined symbols for architecture x86_64:
  "_NSTemporaryDirectory", referenced from:
      NCS::ObjectiveCCPlusPlusWrapper::Utilities::GetTempDirectory() in libNCSEcw.a[50](ObjectiveCCPlusPlusWrapper.o)
  "_NSUserName", referenced from:
      NCS::ObjectiveCCPlusPlusWrapper::Utilities::GetUserName() in libNCSEcw.a[50](ObjectiveCCPlusPlusWrapper.o)
  "_OBJC_CLASS_$_NSProcessInfo", referenced from:
       in libNCSEcw.a[50](ObjectiveCCPlusPlusWrapper.o)
  "___CFConstantStringClassReference", referenced from:
       in libNCSEcw.a[50](ObjectiveCCPlusPlusWrapper.o)
  "_objc_autoreleasePoolPop", referenced from:
      NCS::ObjectiveCCPlusPlusWrapper::Utilities::GetTempDirectory() in libNCSEcw.a[50](ObjectiveCCPlusPlusWrapper.o)
      NCS::ObjectiveCCPlusPlusWrapper::Utilities::GetUserName() in libNCSEcw.a[50](ObjectiveCCPlusPlusWrapper.o)
      NCS::ObjectiveCCPlusPlusWrapper::Utilities::GetCurrentProcessId() in libNCSEcw.a[50](ObjectiveCCPlusPlusWrapper.o)
  "_objc_autoreleasePoolPush", referenced from:
      NCS::ObjectiveCCPlusPlusWrapper::Utilities::GetTempDirectory() in libNCSEcw.a[50](ObjectiveCCPlusPlusWrapper.o)
      NCS::ObjectiveCCPlusPlusWrapper::Utilities::GetUserName() in libNCSEcw.a[50](ObjectiveCCPlusPlusWrapper.o)
      NCS::ObjectiveCCPlusPlusWrapper::Utilities::GetCurrentProcessId() in libNCSEcw.a[50](ObjectiveCCPlusPlusWrapper.o)
  "_objc_msgSend", referenced from:
      NCS::ObjectiveCCPlusPlusWrapper::Utilities::GetTempDirectory() in libNCSEcw.a[50](ObjectiveCCPlusPlusWrapper.o)
      NCS::ObjectiveCCPlusPlusWrapper::Utilities::GetTempDirectory() in libNCSEcw.a[50](ObjectiveCCPlusPlusWrapper.o)
      NCS::ObjectiveCCPlusPlusWrapper::Utilities::GetTempDirectory() in libNCSEcw.a[50](ObjectiveCCPlusPlusWrapper.o)
      NCS::ObjectiveCCPlusPlusWrapper::Utilities::GetUserName() in libNCSEcw.a[50](ObjectiveCCPlusPlusWrapper.o)
      NCS::ObjectiveCCPlusPlusWrapper::Utilities::GetCurrentProcessId() in libNCSEcw.a[50](ObjectiveCCPlusPlusWrapper.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

Lacks linking to `-framework Foundation`

### Steps to reproduce the issue

https://gdal.org/en/stable/drivers/raster/ecw.html#standalone-plugin-compilation

### Versions and provenance

macos 14.6.1; vcpkg; gdal 3.10.0

### Additional context

_No response_

--- a/cmake/modules/packages/FindECW.cmake
+++ b/cmake/modules/packages/FindECW.cmake
@@ -89,11 +89,15 @@ if (ECW_VERSION_STRING VERSION_GREATER_EQUAL 5.5)
             if( "${ECW_LIBRARY}" MATCHES "NCSEcwS" )
                 set(ECW_INTERFACE_COMPILE_DEFINITIONS "${ECW_INTERFACE_COMPILE_DEFINITIONS};NCSECW_STATIC_LIBS")
             endif()
+            if(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+                list(APPEND ECW_EXTRA_LINK_LIBRARIES "-framework Foundation")
+            endif()
             set_target_properties(ECW::ECW_ALL PROPERTIES
                                 INTERFACE_INCLUDE_DIRECTORIES "${ECW_INCLUDE_DIRS}"
                                 INTERFACE_COMPILE_DEFINITIONS "${ECW_INTERFACE_COMPILE_DEFINITIONS}"
                                 IMPORTED_LINK_INTERFACE_LANGUAGES "C"
-                                IMPORTED_LOCATION "${ECW_LIBRARY}")
+                                IMPORTED_LOCATION "${ECW_LIBRARY}"
+                                INTERFACE_LINK_LIBRARIES ${ECW_EXTRA_LINK_LIBRARIES})
         endif()
     endif()
 elseif(ECW_VERSION_STRING VERSION_GREATER_EQUAL 4.0)

